addons/product/tests/test_pricelist.py

```diff
@@ -198,3 +198,22 @@ def test_pricelists_res_partner_form(self):
 
         with Form(partner) as partner_form:
             self.assertEqual(partner_form.property_product_pricelist, self.sale_pricelist_id)
+
+    def test_pricelist_change_to_formula_and_back(self):
+        pricelist_2 = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'item_ids': [
+                Command.create({
+                    'compute_price': 'percentage',
+                    'percent_price': 20,
+                    'base': 'pricelist',  # based on pricelist_1
```

Comment:
Comment no longer makes sense, plus the next line already indicates which base pricelist we're using
```suggestion
                    'base': 'pricelist',
```